bin/pip_find_builddeps.py

```diff
@@ -117,6 +117,26 @@ def generate_file_content(builddeps, is_partial):
     return file_content
 
 
+def _parse_requirements_file(builddeps_file):
+    """Find deps requirements-build.in file."""
+    try:
+        with open(builddeps_file) as f:
+            # ignore line comments or comments added after dependency is declared
+            requirement_re = re.compile(r"^\s*([^\s#]+)")
```

Comment:
Why does the regex allow leading whitespace? And should the character class also exclude `;` like the one in `_filter_builddeps()`?

AFAIK semicolon would be used for environment markers:
```
poetry<1.0; python_version<"3.6"
```

Reply:
Mmmm I didn't think thoroughly about leading whitespaces. Generated build requirements file generated by this script should not have it, so I guess skipping it is ok.

I also didn't consider environment markers. I think this match should reproduce the behavior of  `_filter_builddeps()` .

Updating this accordingly

Reply:
I forgot to push :facepalm: 
sorry for that :sweat_smile: 
```
% git status                                                                  ✭
On branch find-build-deps-enhancements
Your branch is ahead of 'origin/find-build-deps-enhancements' by 1 commit.
  (use "git push" to publish your local commits)
```

Reply:
https://github.com/bruno-fs/cachito/commit/e4c83385a79ec53d81cd06085d7b0ffce3ee289f

Reply:
ugh, it wasn't signed off.

Do you prefer I rebase/squash now?

Reply:
Squashing the fixup into the first commit would be nice, otherwise the 3 commits make sense to me :+1: